tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/management/ActiveMQServerControlTest.java

```diff
@@ -514,8 +514,8 @@ public void testRemoveQueueFilter() throws Exception {
 
       producer.send(m);
 
-      assertNotNull(consumer1.receiveImmediate());
-      assertNotNull(consumer2.receiveImmediate());
+      assertNotNull(consumer1.receive(1000));
```

Comment:
@jsmucr: just a note here. if you intended to assertNotNull, do a receive with a timeout.

if you assertNull, do a receiveImmediate (or receiveNoWait in JMS terms).. or a very short timeout.

assertNotNull(receiveImmediate) is a bit fragile and it may break a test intermittently.

Reply:
I'll keep that in mind. Thank you.